html/captive-portal/lib/captiveportal/PacketFence/Controller/Oauth2.pm

```diff
@@ -193,12 +194,16 @@ sub oauth2Result : Path : Args(1) {
                 # Grab JSON content
                 my $json      = new JSON;
                 my $json_text = $json->decode($response->content());
-                if ($provider eq 'google' || $provider eq 'github' || $provider eq 'facebook') {
+                if ($provider eq 'google' || $provider eq 'facebook') {
                     $pid = $json_text->{email};
                 } elsif ($provider eq 'windowslive'){
                     $pid = $json_text->{emails}->{account};
+                } elsif ($provider eq 'github'){
+                    # github doesn't provide email by default
+                    $pid = $json_text->{login}.'@github';
```

Comment:
The email address would be invalid (I tested). 
Putting the full domain would suggest that's it's a real e-mail address in my opinion.